0000-experiment-dataframe.md

```diff
@@ -0,0 +1,320 @@
+```
+
+Analysis results:
+
+|          | name   | value  | quality | device_components |  backend        | creation_datetime          | experiment | ... |
+|----------|--------|--------|---------|-------------------|-----------------|----------------------------|------------|-----|
+| 17636b47 | T1     | 100e-6 | good    | [Q0]              |  ibm_washington | 2022-11-22 03:58:36.523230 | T1         | ... |
+| 2f50a741 | T2star |  80e-6 | good    | [Q0]              |  ibm_washington | 2022-11-22 03:58:36.523230 | T2Ramsey   | ... |
+
+
+#### Example 3: Repeating the same experiment.
+
+```python
+exp = T1(qubit=0, delays=delays)
+
+trend_data = []
+for _ in range(2):
+  experment_data = exp.run()
+  trend_data.append(experment_data)
+  time.sleep(3600)
+
+# combine trend data
+...
+```
+
+Analysis results (combined):
+
+|          | name   | value  | quality | device_components |  backend        | creation_datetime          | experiment | ... |
+|----------|--------|--------|---------|-------------------|-----------------|----------------------------|------------|-----|
+| 03fe91s1 | T1     | 100e-6 | good    | [Q0]              |  ibm_washington | 2022-11-22 04:01:30.313872 | T1         | ... |
+| 1c4ad2a1 | T1     | 103e-6 | good    | [Q0]              |  ibm_washington | 2022-11-22 05:01:14.123201 | T1         | ... |
+
+
+#### Complement
+
+As these examples show, experimentalist can distinguish the outcomes of batch (with `experiment`) and parallel experiments (with `device_components`). Even though there are multiple entries for the identical parameter (example 3), they have different result IDs and creation date times. In other words, one can sort the entries by `creation_datetime` and get the time-ordered sequence of `value` for the trend analysis:
+
+```python
+sorted_frame = result_dataframe.sort_values(by=["creation_datetime"])
+```
+
+alternatively one can directly visualize the data frame with seaborn:
+
+```python
+import seaborn as sns
+
+sns.relplot(
+    data=result_dataframe,
+    kind="line",
+    x="creation_datetime",
+    y="value",
+)
+```
+
+See pandas documentation for more details.
+
+### B. Introduction of artifacts
+
+Result database allows experimentalists to store `artifact` data per experiment ID. Thus, in QE, we can add `ExperimentData.artifact` to save supplementary data such as raw curve data points. In principle this can be a free-from dictionary, but it would be better to provide the artifact base class and subclasses to support custom JSON serialization to generate the REST API payload and the input data type validation. In the following we describe `CurveAnalysisArtifact` for `CurveAnalysis`. Note that artifact must be the [thread-safe container](https://github.com/Qiskit/qiskit-experiments/blob/b150427f86c74a02d638e06507864c7bb060c27c/qiskit_experiments/database_service/utils.py#L168-L245) subclass. Each analysis class must know the associated artifact subclass to populate the supplementary data.
```

Comment:
Do we definitely want to use JSON for artifacts, or can we support other more efficient file formats too? Eg saving a pandas dataframe directly, or np array, or pickling some things? 

Is there a particular reason we need artifacts to be a thread safe container? These could just be arbitrary things that are generated by an experiement/analysis. eg a plot, a dataframe, a zip file etc. I'm not sure we should make any assumptions about them being thread safe or even safe to modify by other experiments.

Probably we just want a list of supported types for artifacts (like svg, dataframe, array, maybe json/picklable things?)